### PR TITLE
Add more lambda defaults/`params` tests

### PR DESCRIPTION
**Pull request by jjonescz**
Completes the following from the test plan #62485:
- [x] Missing constant value types
- [x] GetSymbolInfo returns delegate type with default parameters
- [x] GetConstantValue on default parameter value
- [x] IOperation symbols reflect default values
- [x] MemberSemanticModel should be parented to the correct outer SemanticModel
  - [x] symbols for locals declared outside of a lambda referenced inside default value should be reference identical to symbols returned by GetDeclaredSymbol for the locals and for references to the locals outside of the lambda
  - [x] binder chain should be shared with member model for the enclosing method body
  - [ ] step through the code to make sure the correct things happen
- [x] SpeculativeSemanticModel to ensure errors from default parameter values are not added to declaration table
- [x] Extract local uses `var`
- [x] Convert to method preserves default values

### CODE REVIEW

File: src/EditorFeatures/CSharpTest2/Recommendations/NullKeywordRecommenderTests.cs
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest2/Recommendations/NullKeywordRecommenderTests.cs:323]: we have an equivalent params test for this case, right?

**Comment by CyrusNajmabadi** [comment]
@jaredpar @arkalyanms  IDE signs off on min-bar for lambda defaults/params with this PR.  Thanks!

File: src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs
Comment (jcouv) [inline src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs:13607]: Do we already have a test showing that a different default value prevents sharing of synthesized delegate types? Consider add `var lam5 = (int a = 42) => a;` to this test.

File: src/EditorFeatures/CSharpTest2/Recommendations/DefaultKeywordRecommenderTests.cs
Comment (jcouv) [inline src/EditorFeatures/CSharpTest2/Recommendations/DefaultKeywordRecommenderTests.cs:327]: Consider also testing `var lambda = (int i = $$` (should offer `default` and `null`)

File: src/Workspaces/CSharpTest/Formatting/FormattingTests.cs
Comment (jcouv) [inline src/Workspaces/CSharpTest/Formatting/FormattingTests.cs:10683]: In addition to testing the IDE formatter, we should also cover the compiler `NormalizeWhitespace` if not done already.

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 4) with test suggestions to consider

**Comment by jcouv** [comment]
@jjonescz In addition to the IDE unittests, I would recommend loading your local build of roslyn into Visual Studio ([instructions](https://github.com/dotnet/roslyn/blob/main/docs/contributing/Building%2C%20Debugging%2C%20and%20Testing%20on%20Windows.md#deploying-with-command-line-recommended-method)) and spend a couple of minutes typing code using the new feature.

File: src/EditorFeatures/CSharpTest2/Recommendations/NullKeywordRecommenderTests.cs
Comment (jjonescz) [reply src/EditorFeatures/CSharpTest2/Recommendations/NullKeywordRecommenderTests.cs:323]: Yes, it's already in the feature branch:

https://github.com/dotnet/roslyn/blob/b2325a6242c6beec480b23e9eee603b5db310cfb/src/EditorFeatures/CSharpTest2/Recommendations/ParamsKeywordRecommenderTests.cs#L223-L228

File: src/EditorFeatures/CSharpTest2/Recommendations/DefaultKeywordRecommenderTests.cs
Comment (jjonescz) [reply src/EditorFeatures/CSharpTest2/Recommendations/DefaultKeywordRecommenderTests.cs:327]: Thanks, added: 87514b64a54bae17e8b770127e80b15fd8e67360

File: src/Workspaces/CSharpTest/Formatting/FormattingTests.cs
Comment (jjonescz) [reply src/Workspaces/CSharpTest/Formatting/FormattingTests.cs:10683]: Thanks, added: 2ef39d291d5e1ee1afb8aa3531eb331e6baa8fbe

**Review by jcouv — APPROVED** [review]
Still LGTM Thanks (iteration 6)

**Comment by jcouv** [comment]
@jjonescz I think this change can be merged. The IDE test changes were approved, the compiler test changes only require one review as well, and the only compiler change in a product file is a minor comment change. So I don't think this needs a second review.

**Pull request merged**
